src/Npgsql/NpgsqlActivitySource.cs

```diff
@@ -20,19 +21,56 @@ static NpgsqlActivitySource()
 
     internal static bool IsEnabled => Source.HasListeners();
 
-    internal static Activity? CommandStart(NpgsqlConnector connector, string sql)
+    internal static Activity? CommandStart(NpgsqlConnector connector, string commandText, CommandType commandType)
     {
         var settings = connector.Settings;
-        var activity = Source.StartActivity(settings.Database!, ActivityKind.Client);
+
+        var dbName = settings.Database ?? connector.InferredUserName;
+        string? dbOperation = null;
+        string? dbSqlTable = null;
+        string activityName;
+        switch (commandType)
+        {
+        case CommandType.StoredProcedure:
+            dbOperation = "CALL";
```

Comment:
This might not be a CALL if Npgsql.EnableStoredProcedureCompatMode AppContext switch is on.

Reply:
Good catch. Thanks! Will fix.